presto-main/src/main/java/com/facebook/presto/operator/scalar/TryFunction.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.util.Reflection;
+import com.google.common.base.Functions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Primitives;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.typeParameter;
+import static java.lang.invoke.MethodHandles.catchException;
+import static java.lang.invoke.MethodHandles.constant;
+import static java.lang.invoke.MethodHandles.dropArguments;
+
+public class TryFunction
+        extends SqlScalarFunction
+{
+    public static final TryFunction TRY = new TryFunction();
+
+    public TryFunction()
+    {
+        super("TRY", ImmutableList.of(typeParameter("T")), "T", ImmutableList.of("T"));
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "";
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type innerFunctionReturnType = types.get("T");
+
+        Class<?> returnType = Primitives.wrap(innerFunctionReturnType.getJavaType());
+        MethodHandle exceptionHandler = dropArguments(constant(returnType, null), 0, RuntimeException.class);
```

Comment:
The effect of this line is returning null if any exception is encountered? Add test for this.